lib/Inspection/VPack.h

```diff
@@ -42,6 +42,13 @@ void serialize(Builder& builder, T& value) {
   }
 }
 
+template<class T>
+[[nodiscard]] auto serialize(T& value) -> std::shared_ptr<Builder> {
+  auto builder = std::make_shared<Builder>();
+  serialize(*builder, value);
+  return builder;
+}
```

Comment:
Any particular reason why we return a shared_ptr instead of a Builder?

Reply:
Builder are pretty expensive object to move around. However, I would suggest to use something else than a builder entirely. Something that still owns the slice.

Reply:
Returning a builder was the obvious thing that I knew to return; as Lars says, builders are fairly heavy, so a pointer seemed like not the worst idea.

Then I thought about having a structure that owns a slice (or a builder), would https://github.com/arangodb/velocypack/blob/main/include/velocypack/SliceContainer.h be the correct thing to use?

Reply:
Looking at this again
 * the formatter overload should be for a velocypack slice,
 * serialize could return just a SliceContainer because we don't really want to be able to fiddle about with that slice again, anyway.